imi/src/main/java/org/motechproject/nms/imi/service/impl/CdrFileServiceImpl.java

```diff
@@ -121,18 +200,27 @@ private void parseDetailRecord(CallDetailRecordDto cdr, Map<String, CallSummaryR
     }
 
 
-    // Create an in-memory set of summary records from all detail records
-    private ParseResults parseDetailRecords(File file, FileInfo fileInfo) {
+    private int getMaxErrorCount() {
+        try {
+            return Integer.parseInt(settingsFacade.getProperty(MAX_CDR_ERROR_COUNT));
+        } catch (NumberFormatException e) {
+            return MAX_CDR_ERROR_COUNT_DEFAULT;
+        }
+    }
+
+
+    @Override
+    public VerifyResults aggregateDetailFile(File file, FileInfo fileInfo, boolean verifyOnly) {
```

Comment:
Thinking about this some more, and looking at the calling code instead of a boolean what about an enum of actions?

http://stackoverflow.com/questions/6282619/is-there-a-way-to-bitwise-or-enums-in-java

So instead of calling code like this:

```
aggregateDetailFile(file, fileInfo, false);
```

You can do this

```
aggregateDetailFile(file, fileInfo, EnumSet.of(Action.AGGREGATE, Action.VERIFY));
```

or

```
aggregateDetailFile(file, fileInfo, EnumSet.of(Action.VERIFY));
```

Then the signature would be:

```
public VerifyResults aggregateDetailFile(File file, FileInfo fileInfo, Set<Actions> actions)
```

and the check would be:

```
if (actions.contains(Actions.AGGREGATE) 
```

Not as concise as a bitflag but not terrible, and most likely not needed.  I just went down a rat hole, it's dark and scary in here.